examples/tensorflow2/scripts/tf_keras_fit_hvd.py

```diff
@@ -0,0 +1,146 @@
+"""
+This script is a simple MNIST training script which uses Horovod and Tensorflow 2.x Keras interface.
+It is designed to be used with SageMaker Debugger in an official SageMaker Framework container (i.e. AWS Deep Learning Container).
+You will notice that this script looks exactly like a normal TensorFlow training script.
+The hook needed by SageMaker Debugger to save tensors during training will be automatically added in those environments.
+The hook will load configuration from json configuration that SageMaker will put in the training container from the
+configuration provided using the SageMaker python SDK when creating a job.
+For more information, please refer to https://github.com/awslabs/sagemaker-debugger/blob/master/docs/sagemaker.md
+
+This script has been adapted from an example in Horovod repository https://github.com/uber/horovod
+"""
+
+# Standard Library
+import argparse
+from datetime import datetime
+
+# Third Party
+import horovod.tensorflow.keras as hvd
+import tensorflow.compat.v2 as tf
+
+
+def str2bool(v):
+    if isinstance(v, bool):
+        return v
+    if v.lower() in ("yes", "true", "t", "y", "1"):
+        return True
+    elif v.lower() in ("no", "false", "f", "n", "0"):
+        return False
+    else:
+        raise argparse.ArgumentTypeError("Boolean value expected.")
+
+
+def get_data(batch_size):
+    (mnist_images, mnist_labels), _ = tf.keras.datasets.mnist.load_data(
+        path="mnist-%d.npz" % hvd.rank()
+    )
+
+    dataset = tf.data.Dataset.from_tensor_slices(
+        (
+            tf.cast(mnist_images[..., tf.newaxis] / 255.0, tf.float32),
+            tf.cast(mnist_labels, tf.int64),
+        )
+    )
+    dataset = dataset.repeat().shuffle(10000).batch(batch_size)
+    return dataset
+
+
+def get_model():
+    mnist_model = tf.keras.Sequential(
+        [
+            tf.keras.layers.Conv2D(32, [3, 3], activation="relu", input_shape=(28, 28, 1)),
+            tf.keras.layers.Conv2D(64, [3, 3], activation="relu"),
+            tf.keras.layers.MaxPooling2D(pool_size=(2, 2)),
+            tf.keras.layers.Dropout(0.25),
+            tf.keras.layers.Flatten(),
+            tf.keras.layers.Dense(128, activation="relu"),
+            tf.keras.layers.Dropout(0.5),
+            tf.keras.layers.Dense(10, activation="softmax"),
+        ]
+    )
+    return mnist_model
+
+
+def train(model, dataset, epoch):
+    # Horovod: Specify `experimental_run_tf_function=False` to ensure TensorFlow
+    # uses hvd.DistributedOptimizer() to compute gradients.
+    model.compile(
+        loss=tf.losses.SparseCategoricalCrossentropy(),
+        optimizer=opt,
+        metrics=["accuracy"],
+        experimental_run_tf_function=False,
+    )
+
```

Comment:
Can we test the behavior by parameterizing the `run_eagerly` and `eager` flags?

Reply:
due to this issue with TF 2.2 [Reference_1](https://stackoverflow.com/questions/58565913/typeerror-an-op-outside-of-the-function-building-code-is-being-passed-a-graph-t), [Reference_2](https://github.com/tensorflow/probability/issues/620), I can't add non-eager testing at this point.

This issue doesn't occur with TF 2.3.0rc0. I'll create a follow up item to get back to this once TF 2.3 is released.

Reply:
https://github.com/awslabs/sagemaker-debugger/issues/286